src/plugins/data/common/es_query/es_query/types.ts

```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { IFieldType } from '../../index_patterns';
+
+export interface MinimalIndexPattern {
```

Comment:
@mattkime do you have a better name for this?
And what do you think in general?

Reply:
I think this is fine. Its a bit hacky but it solves the problem without increasing complexity.